src/sync/sync-api.js

```diff
@@ -125,34 +129,87 @@ export class SyncApi extends TypedEmitter {
    */
   #getState(namespaceSyncState) {
     const state = reduceSyncState(namespaceSyncState)
-    state.data.syncing = this.#isSyncing
+
+    state.data.syncing =
+      this.#wantsToSyncData &&
+      (!this.#isBackgrounded ||
+        !isSynced(namespaceSyncState, 'full', this.#peerSyncControllers))
+
```

Comment:
It probably makes sense to rename `state['data' | 'initial'].syncing` to `state['data' | 'initial'].syncEnabled` to be clear about what this means.

Reply:
This is essentially calculating the value of `syncEnabledState` based on the same logic that is in `#updateState()`, and we need to remember to keep those two pieces of logic in sync. Hoisting syncEnabledState to somewhere that can be read here would avoid having this duplication.

For completeness we should also set `state.initial.syncing` here, because currently it just defaults to always be `true`, but now it can be false.

Reply:
Refactored in 8fe34aaec18b07facd878c7c93fe96c67d8295a8.